R/dataframe-frame.R

```diff
@@ -2256,3 +2256,90 @@ dataframe__count <- function() {
   self$lazy()$count()$collect(`_eager` = TRUE) |>
     wrap()
 }
+
+#' Show a dense preview of the DataFrame
+#'
+#' The formatting shows one line per column so that wide DataFrames display
+#' cleanly. Each line shows the column name, the data type, and the first few
+#' values.
+#'
+#' @inheritParams rlang::args_dots_empty
+#' @param max_items_per_column Maximum number of items to show per column.
+#' @param max_colname_length Maximum length of the displayed column names.
+#' Values that exceed this value are truncated with a trailing ellipsis.
+#'
+#' @return Returns a character value (invisibly)
+#'
+#' @examples
+#' df <- as_polars_df(iris)
+#' df$glimpse()
+#'
+#' df$glimpse(max_items_per_column = 3)
+#'
+#' df$glimpse(max_items_per_column = 3, max_colname_length = 3)
+dataframe__glimpse <- function(
+  ...,
+  max_items_per_column = 10,
+  max_colname_length = 50
+) {
+  wrap({
+    check_dots_empty0(...)
+    max_n_values <- min(max_items_per_column, self$height)
+    schema <- self$schema
+
+    parse_column <- \(col_name, dtype) {
+      values <- self$select(pl$col(col_name)$slice(0, max_n_values)) |>
+        as.list()
+      val_str <- toString(values[[1]])
```

Comment:
(I have not checked the Python Polars implementation.)

Unexpected results may occur because in the future it may be possible to change the default values of arguments in the `as.list(<DataFrame>)` with the options function, similar to the current r-polars.
Wouldn't it be better to use cast in Polars to make it into a string and then return it to the R side?

Reply:
Ah yes, that's a good point. I've added a cast to String which slightly changes the output (e.g "5" is replaced by "5.0") but I think it's fine.

I've also added an int64 column in the snapshots to catch this kind of changes in `as.list()`.

Reply:
I keep the int8 column on purpose in snapshots to check the printing of abbreviated data types that don't have the same length.